docs/protocol.md

```diff
@@ -1,26 +1,83 @@
-# TChannel v2
+# Specification
 
-## Design goals
+## Introduction
+
+### Problem Statement
+
+TChannel was developed at Uber during a period of explosive growth in its
+[service oriented architecture][soa]. Services living in a large, distributed
+system like this usually encounter similar classes of problems, such as:
```

Comment:
run on sentence, replace this comma with something ilke "problems.  Some of the problems addressed:"